src/client/components/home/Home.js

```diff
@@ -1,28 +1,89 @@
 import { makeStyles } from '@material-ui/core';
 import Typography from '@material-ui/core/Typography';
-import React, { useContext } from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { APP_AUTHOR, APP_DESCRIPTION, APP_NAME } from '../../config/constants';
 import CollectionsGrid from '../collection/CollectionsGrid';
-import { CollectionContext } from '../CollectionProvider';
 import Seo from '../common/Seo';
+import { useCollections } from '../../utils/collections';
+import Loader from '../common/Loader';
+import Search from './Search';
 
-const useStyles = makeStyles(() => ({
+const useStyles = makeStyles((theme) => ({
   wrapper: {
     padding: '4vw',
   },
+  root: {
+    padding: theme.spacing(0.25, 0.5),
+    margin: '100px 0px',
+    display: 'flex',
+    alignItems: 'center',
+    width: '100%',
+  },
+  input: {
+    marginLeft: '20px',
+    flex: 6,
+  },
+  iconButton: {
+    padding: 10,
+  },
+  divider: {
+    height: 28,
+    margin: 4,
+  },
+  typographyMargin: {
+    margin: '10px 0px',
+  },
 }));
 
 function Home() {
   const { t } = useTranslation();
-  const { collections } = useContext(CollectionContext);
+  const [queried, setQueried] = useState(false);
+  const [searchResults, setSearchResults] = useState(null);
+  const { collections, isLoading } = useCollections();
+
+  const handleSearch = (event) => {
+    const query = event.target.value.trim().toLowerCase();
+    if (query.length > 0) {
+      setSearchResults(
+        collections.filter(
+          (collection) =>
+            collection.name.toLowerCase().includes(query) ||
+            collection.author?.name.toLowerCase().includes(query),
+        ),
+      );
+      setQueried(true);
+    }
+  };
   const classes = useStyles();
   return (
     <>
       <Seo title={APP_NAME} description={APP_DESCRIPTION} author={APP_AUTHOR} />
       <div className={classes.wrapper}>
-        <Typography variant="h2">{t('Home')}</Typography>
-        <CollectionsGrid collections={collections} />
+        <Typography variant="h3" align="center">
+          {t('Graasp Collections Directory')}
+        </Typography>
+        <Search handleSearch={handleSearch} isLoading={isLoading} />
+        {isLoading ? (
+          <Loader />
+        ) : queried ? (
+          <>
+            <Typography variant="h3" className={classes.typographyMargin}>
+              {t('Search Results')}
+            </Typography>
+            {searchResults.length > 0 ? (
+              <CollectionsGrid collections={searchResults} />
+            ) : (
+              <Typography variant="body1" className={classes.typographyMargin}>
+                {t('No results found.')}
+              </Typography>
```

Comment:
Maybe factor out some components in render functions. Eg. `queries ? renderResults() : renderNoResult() `